src/awesome/client.rs

```diff
@@ -42,9 +43,60 @@ impl<'lua> ToLua<'lua> for Client<'lua> {
     }
 }
 
-impl UserData for ClientState {}
+impl UserData for ClientState {
+    fn add_methods(methods: &mut UserDataMethods<Self>) {
+        methods.add_method("geometry", |lua, state, (geometry,): (Value,)| {
```

Comment:
If you want to add methods on the object, I suggest you use the [same API we use else where](https://github.com/way-cooler/way-cooler/blob/master/src/awesome/drawable.rs#L120)

Reply:
That sets it on the global. Seems like the real api is here:

https://github.com/way-cooler/way-cooler/blob/833db6ad5cbd040f3b124818c65b4fab24d0d729/src/awesome/drawable.rs#L35-L44

Do you really think that's better?

Reply:
Well that code adds it to the meta table whereas this adds it to the user data AFAIK. So it's a little different.

For our purposes they _should_ be be equivalent AFAIK? We should probably standardize on one way though. CC @psychon if it makes a difference?

@acrisci Keep it the way it is for now if it works.

Reply:
Well... dunno. I copied the approach for the code above from other places of way-cooler, but I vaguely remember that there were some inconsistencies.

Also, this `add_methods` thing comes from rlua, not lua. It is implemented via an `__index` metamethod as well: https://github.com/chucklefish/rlua/blob/71f3dd50a1f396ea2ff36462e3deb6a48a132c43/src/lua.rs#L814

So, I guess way-cooler grew a reimplementation of some rlua feature somewhere along the way...

Reply:
acrisci's implementation puts the methods on the user data and my object implementation puts it on the table associated with the user data IIRC. The default index method ensures that our functions are called if defined on the table.